app/models/tagging_spreadsheet.rb

```diff
@@ -1,5 +1,8 @@
 class TaggingSpreadsheet < ActiveRecord::Base
   validates :url, presence: true
+  validates_presence_of :state
+  validates_inclusion_of :state, in: %w(uploaded errored ready_to_import imported)
```

Comment:
Nice! If you wanted to use a state machine, I've tried this in the past and really liked it https://github.com/aasm/aasm